core/trino-main/src/main/java/io/trino/execution/scheduler/SqlQueryScheduler.java

```diff
@@ -1577,6 +1575,10 @@ public void schedule()
                         ScheduleResult result = stageSchedulers.get(stageExecution.getStageId())
                                 .schedule();
 
+                        if (stateMachine.getState() == DistributedStagesSchedulerState.PLANNED && stageExecution.getAllTasks().size() > 0) {
```

Comment:
same here: `if (stateMachine.getState() == DistributedStagesSchedulerState.PLANNED` is not needed

Reply:
I wanted to avoid calling `stageExecution.getAllTasks()` every time as it has to acquire a list of tasks under a lock and made the similar code in the coordinator scheduler consistent with this. Likely it won't be a problem, but at the same time I wonder if it introduces enough confusion to justify extra CPU cycles?

Reply:
I see. Keep it then